app/controllers/checkout_controller.rb

```diff
@@ -46,7 +46,7 @@ def update
     params_adapter = Checkout::FormDataAdapter.new(permitted_params, @order, spree_current_user)
     return action_failed unless @order.update(params_adapter.params[:order] || {})
 
-    checkout_workflow(params_adapter.shipping_method_id)
+    checkout_workflow(params_adapter.shipping_method_id, params[:advance_to_state])
```

Comment:
If we don't subclass the current checkout controller we can leave it completely untouched whilst we work on this and then we don't have to worry about testing it (or breaking anything) :+1: